src/exception/api.error.js

```diff
@@ -0,0 +1,31 @@
+export class ApiError extends Error {
+  constructor({ message, status, errors = {} }) {
+    super(message);
+
+    this.status = status;
+    this.errors = errors;
+  }
+
+  static badRequest(message, errors) {
+    return new ApiError({
+      message,
+      errors,
+      status: 400,
+    });
+  }
+
+  static unautorized(errors) {
+    return new ApiError({
+      message: 'unautirized user',
+      errors,
+      status: 401,
+    });
+  }
```

Comment:
There is a typo in the method name 'unautorized'. It should be 'unauthorized' to correctly reflect the standard spelling.